gama/genetic_programming/operations.py

```diff
@@ -1,44 +1,323 @@
+) -> Union[str, None]:
+    """Extract valid hyperparameters from a condition.
+
+    For each supported ConfigSpace condition type, extract the valid hyperparameters
+    from the condition. The valid hyperparameters are the hyperparameters that are
+    valid for the given condition and configuration based on the meta_key.
+
+    Supported ConfigSpace condition types:
+    - EqualsCondition
+    - AndConjunction
+
+    Readers are encouraged to add support for more ConfigSpace condition types if
+    needed. Open an issue or pull request on the GAMA GitHub repository.
+
+    Parameters
+    ----------
+    cond : cs.conditions
+        A condition of type ConfigSpace.
+    config : cs.Configuration
+        A configuration of type ConfigSpace.
+    config_meta : dict
+        The meta information of the ConfigSpace.
+    meta_key : str
+        The meta key of the ConfigSpace.
+    """
+    if meta_key not in config_meta:
+        raise ValueError(f"Meta key {meta_key} not in config_meta")
+    if type(cond) not in [cs.conditions.EqualsCondition, cs.conditions.AndConjunction]:
+        raise ValueError(
+            f"Condition type {type(cond)} not supported. Refer to "
+            f"docstring for supported condition types."
+        )
+    if isinstance(cond, cs.conditions.EqualsCondition):
+        if (
+            cond.parent.name == config_meta[meta_key]
+            and cond.value == config[config_meta[meta_key]]
+        ):
+            return cond.child.name
+    elif isinstance(cond, cs.conditions.AndConjunction):
+        for component in cond.components:
+            if (
+                component.parent.name == config_meta[meta_key]
+                and component.value == config[config_meta[meta_key]]
+            ):
+                return component.child.name
+    return None
+
+
+def _config_estimator_to_primitive_node(
+    config: cs.Configuration,
+    config_meta: dict,
+    conditions: List[Any],
+    output_type: Optional[str] = None,
+) -> Tuple[Primitive, List[Terminal]]:
+    """Create a PrimitiveNode from a configuration of type ConfigSpace (estimator).
+
+    Creates a PrimitiveNode from a configuration of type ConfigSpace. Focuses on
+    the estimator part of the configuration. It starts by creating a Primitive for
+    the selected estimator. Then it determines the valid hyperparameters for the
+    estimator based on the conditions. Finally, it creates a Terminal for each valid
+    hyperparameter for the estimator.
+
+    Parameters
+    ----------
+    config : cs.Configuration
+        A configuration of type ConfigSpace.
+    config_meta : dict
+        The meta information of the ConfigSpace.
+    conditions : List[Any]
+        The conditions of the ConfigSpace.
+    output_type : str, optional
+        The output type of the PrimitiveNode, by default None.
+    """
+    if (
+        "estimators" not in config_meta
+        or config_meta["estimators"] not in config.get_dictionary()
+    ):
```

Comment:
There are a _lot_ of checks all over the place on these kind of memberships for config meta and so on. With a check up front, can't we get rid of all of these? If there does happen to be a mistake somehow, it should throw a hard error regardless, right?